_includes/releases/v22.1/v22.1.0-alpha.2.md

```diff
@@ -0,0 +1,640 @@
+- Implemented the [`ALTER TABLE ... RESET (...)`](../v22.1/alter-table.html) syntax. This statement currently does not change the schema. [#75429][#75429]
+- S3 URIs used for [`BACKUP`](../v22.1/backup.html), [`EXPORT`](../v22.1/export.html), or [`CHANGEFEED`](../v22.1/use-changefeeds.html) can now include the query paramter `S3_STORAGE_CLASS` to configure the storage class used when that job creates objects in the designated S3 bucket. [#75588][#75588]
+- The [cost based optimizer](../v22.1/cost-based-optimizer.html) now modifies the query cost based on the `avg_size` table statistic, which may change query plans. This is controlled by the [session variable](../v22.1/set-vars.html) `cost_scans_with_default_col_size`, and can be disabled by setting it to `true`: `SET cost_scans_with_default_col_size=true`. [#74551][#74551]
+- The [`crdb_internal.jobs`](../v22.1/crdb-internal.html) table now has a new column `execution_events` which is a structured JSON form of `execution_errors`. [#75556][#75556]
+- The [privileges](../v22.1/security-reference/authorization.html) reported in `information_schema.schema_privileges` for non-user-defined schemas no longer are inferred from the privileges on the parent database. Instead, virtual schemas (like `pg_catalog` and `information_schema`) always report the `USAGE` privilege for the public role. The `pg_temp` schema always reports `USAGE` and `CREATE` privileges for the public role. [#75628][#75628]
+- Transaction ID to transaction fingerprint ID mapping is now stored in the new transaction ID cache, a FIFO unordered in-memory buffer. The size of the buffer is 64 MB by default and configurable via `sql.contention.txn_id_cache.max_size` [cluster setting](../v22.1/cluster-settings.html). Consequentially, two additional metrics are introduced:
+    - `sql.contention.txn_id_cache.size`: the current memory usage of transaction ID cache.
+    - `sql.contention.txn_id_cache.discarded_count`: the number of resolved  transaction IDs that are dropped due to memory constraints. [#74115][#74115]
+- Added new [builtin functions](../v22.1/functions-and-operators.html#built-in-functions) called `crdb_internal.revalidate_unique_constraint`, `crdb_internal.revalidate_unique_constraints_in_table`, and `crdb_internal.revalidate_unique_constraints_in_all_tables`, which can be used to revalidate existing unique constraints. The different variations support validation of a single constraint, validation of all unique constraints in a table, and validation of all unique constraints in all tables in the current database, respectively. If any constraint fails validation, the functions will return an error with a hint about which data caused the constraint violation. These violations can then be resolved manually by updating or deleting the rows in violation. This will be useful to users who think they may have been affected by issue [#73024](https://github.com/cockroachdb/cockroach/issues/73024). [#75548][#75548]
```

Comment:
```suggestion
- Added new [built-in functions](../v22.1/functions-and-operators.html#built-in-functions) called `crdb_internal.revalidate_unique_constraint`, `crdb_internal.revalidate_unique_constraints_in_table`, and `crdb_internal.revalidate_unique_constraints_in_all_tables`, which can be used to revalidate existing unique constraints. The different variations support validation of a single constraint, validation of all unique constraints in a table, and validation of all unique constraints in all tables in the current database, respectively. If any constraint fails validation, the functions will return an error with a hint about which data caused the constraint violation. These violations can then be resolved manually by updating or deleting the rows in violation. This will be useful to users who think they may have been affected by issue [#73024](https://github.com/cockroachdb/cockroach/issues/73024). [#75548][#75548]
```